src/server/game/Anticheat/AnticheatMgr.h

```diff
@@ -0,0 +1,151 @@
+/*
+* Copyright (C) 2011-2016 TrinityCore <http://www.trinitycore.org/>
+*
+* This program is free software; you can redistribute it and/or modify it
+* under the terms of the GNU General Public License as published by the
+* Free Software Foundation; either version 2 of the License, or (at your
+* option) any later version.
+*
+* This program is distributed in the hope that it will be useful, but WITHOUT
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+* more details.
+*
+* You should have received a copy of the GNU General Public License along
+* with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef SC_ACMGR_H
+#define SC_ACMGR_H
+
+#include "AnticheatData.h"
+#include "Common.h"
+#include "SharedDefines.h"
+#include "ScriptMgr.h"
+#include "Player.h"
+
+class ChatHandler;
+class AnticheatCheck;
+
+enum DetectionTypes
+{
+    SPEED_HACK_DETECTION            = 0x01,
+    FLY_HACK_DETECTION              = 0x02,
+    WALK_WATER_HACK_DETECTION       = 0x04,
+    JUMP_HACK_DETECTION             = 0x08,
+    TELEPORT_PLANE_HACK_DETECTION   = 0x10,
+    CLIMB_HACK_DETECTION            = 0x20
+};
+
+typedef std::unordered_map<ObjectGuid, AnticheatData> AnticheatPlayersDataMap;
+
+class AnticheatMgr
+{
+    friend class AnticheatCheck;
+
+    AnticheatMgr();
+    ~AnticheatMgr();
+
+    public:
+        static AnticheatMgr* instance()
+        {
+           static AnticheatMgr instance;
```

Comment:
please keep an eye on TC development and apply the latest standard, for example https://github.com/TrinityCore/TrinityCore/commit/b7f936891aeb86eeb4eb67caa78e672858aa1e20